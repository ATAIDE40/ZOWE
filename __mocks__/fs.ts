/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

const __mocks__path = require("path");

let __mocks__fs = jest.genMockFromModule("fs") as any;
const __mocks__oldReadFileSync = require.requireActual("fs").readFileSync;

// This is a custom function that our tests can use during setup to specify
// what the files on the "mock" filesystem should look like when any of the
// `fs` APIs are used.
let __mocks__mockFiles = Object.create(null);
function __mocks__fs____setMockFiles(newMockFiles: { [key: string]: string }) {
    __mocks__mockFiles = Object.create(null);
    for (const file in newMockFiles) {
        const dir = __mocks__path.dirname(file);
        if (!__mocks__mockFiles[dir]) {
            __mocks__mockFiles[dir] = {};
        }
        __mocks__mockFiles[dir][__mocks__path.basename(file)] = newMockFiles[file];
    }
}

// A custom version of `readdirSync` that reads from the special mocked out
// file list set via __setMockFiles
function __mocks__fs__readdirSync(filePath: string) {
    return __mocks__mockFiles[filePath] || {};
}

// A custom version of `existsSync` that reads from the special mocked out
// file list set via __setMockFiles
function __mocks__fs__existsSync(filePath: string) {
    const fileContents = __mocks__fs__readdirSync(__mocks__path.dirname(filePath))[__mocks__path.basename(filePath)];
    if (typeof fileContents === "undefined") {
        return false;
    }
    return true;
}

// A custom version of `readFileSync` that reads from the special mocked out
// file list set via __setMockFiles
function __mocks__fs__readFileSync(filePath: string, encoding?: string) {
    // Don't mock if yargs is trying to load a locale json file
    if (filePath.match(/node_modules.yargs/)) {
        return __mocks__oldReadFileSync(filePath, encoding);
    }

    if (!__mocks__fs__existsSync(filePath)) {
        throw new Error("File not found");
    }
    return __mocks__fs__readdirSync(__mocks__path.dirname(filePath))[__mocks__path.basename(filePath)];
}

// A custom version of `lstatSync` that reads from the special mocked out
// file list set via __setMockFiles
function __mocks__fs__lstatSync(filePath: string) {
    return {
        isFile: () => false
    };
}

__mocks__fs.__setMockFiles = __mocks__fs____setMockFiles;
__mocks__fs.existsSync = __mocks__fs__existsSync;
__mocks__fs.readFileSync = __mocks__fs__readFileSync;
__mocks__fs.readdirSync = __mocks__fs__readdirSync;
__mocks__fs.lstatSync = __mocks__fs__lstatSync;

module.exports = __mocks__fs;
