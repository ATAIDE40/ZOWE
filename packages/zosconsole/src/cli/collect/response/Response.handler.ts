/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { ICommandHandler, IHandlerParameters, Session } from "@brightside/imperative";
import { CollectCommand, ICollectParms, IConsoleResponse } from "../../../../../zosconsole";

/**
 * Handle to collect a MVS console command response
 * @export
 * @class Handler
 * @implements {ICommandHandler}
 */
export default class Handler implements ICommandHandler {

    public async process(commandParameters: IHandlerParameters) {

        const profile = commandParameters.profiles.get("zosmf");
        const session = new Session({
            type: "basic",
            hostname: profile.host,
            port: profile.port,
            user: profile.user,
            password: profile.pass,
            base64EncodedAuth: profile.auth,
            rejectUnauthorized: profile.rejectUnauthorized,
        });
        const collectParms: ICollectParms = {
            commandResponseKey: commandParameters.arguments.responsekey,
            consoleName: commandParameters.arguments["console-name"],
        };

        const response: IConsoleResponse = await CollectCommand.collect(session, collectParms);

        // Print out the response
        commandParameters.response.console.log(response.commandResponse);

        // Return as an object when using --response-format-json
        commandParameters.response.data.setObj(response);

    }
}
