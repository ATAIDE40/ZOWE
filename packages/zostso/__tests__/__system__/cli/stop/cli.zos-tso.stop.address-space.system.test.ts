/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { ITestEnvironment } from "../../../../../../__tests__/__src__/environment/doc/response/ITestEnvironment";
import * as fs from "fs";
import { TestEnvironment } from "../../../../../../__tests__/__src__/environment/TestEnvironment";
import { runCliScript } from "../../../../../../__tests__/__src__/TestUtils";
import { TestProperties } from "../../../../../../__tests__/__src__/properties/TestProperties";
import { ITestSystemSchema } from "../../../../../../__tests__/__src__/properties/ITestSystemSchema";

// Test Environment populated in the beforeAll();
let TEST_ENVIRONMENT: ITestEnvironment;
describe("zos-tso start address-space", () => {

    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENVIRONMENT = await TestEnvironment.setUp({
            testName: "zos_tso_start_as",
            tempProfileTypes: ["zosmf", "tso"]
        });
    });
    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
    });

    it("should display the help", async () => {
        const response = runCliScript(__dirname + "/__scripts__/address-space/address_space_help.sh", TEST_ENVIRONMENT);
        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
        expect(response.stdout.toString()).toMatchSnapshot();
    });

    it("should throw an error if servlet key parameter is not provided", async () => {
        const response = runCliScript(__dirname + "/__scripts__/address-space/as_error_stop.sh", TEST_ENVIRONMENT);
        expect(response.status).toBe(1);
        expect(response.stdout.toString()).toBe("");
        expect(response.stderr.toString()).toMatchSnapshot();
    });

    it("should throw an error if provided address space is inactive", async () => {
        const response = runCliScript(__dirname + "/__scripts__/address-space/as_error_stop.sh", TEST_ENVIRONMENT);
        expect(response.status).toBe(1);
        expect(response.stdout.toString()).toBe("");
        expect(response.stderr.toString()).toMatchSnapshot();
    });

    it("should successfully issue the command", async () => {
        const regex = fs.readFileSync(__dirname + "/__regex__/as_stop_response.regex").toString();
        const response = runCliScript(__dirname + "/__scripts__/address-space/as_success_stop.sh", TEST_ENVIRONMENT);
        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
        expect(new RegExp(regex, "g").test(response.stdout.toString())).toBe(true);
    });

    it("should complete the command successfully and return a valid JSON response", async () => {
        const regex = fs.readFileSync(__dirname + "/__regex__/as_stop_response.regex").toString();
        const response = runCliScript(__dirname + "/__scripts__/address-space/as_success_stop_rfj.sh", TEST_ENVIRONMENT);
        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);

        // Convert response to an object and check fields
        const respObj = JSON.parse(response.stdout.toString());
        expect(respObj.success).toBe(true);
        expect(respObj.message).toBe("");
        expect(respObj.stderr).toBe("");
        expect(respObj.error).toBeUndefined();
        expect(respObj.data.servletKey).toBeDefined();
        expect(new RegExp(regex, "g").test(response.stdout.toString())).toBe(true);
    });

    describe("without profiles", () => {

        // Create a separate test environment for no profiles
        let TEST_ENVIONMENT_NO_PROF: ITestEnvironment;
        let DEFAULT_SYSTEM_PROPS: ITestSystemSchema;

        beforeAll(async () => {
            TEST_ENVIONMENT_NO_PROF = await TestEnvironment.setUp({
                testName: "zos_tso_stop_as_without_profiles"
            });

            const sysProps = new TestProperties(TEST_ENVIONMENT_NO_PROF.systemTestProperties);
            DEFAULT_SYSTEM_PROPS = sysProps.getDefaultSystem();
        });

        afterAll(async () => {
            await TestEnvironment.cleanUp(TEST_ENVIONMENT_NO_PROF);
        });

        it("should successfully issue the command without a profile", async () => {
            const regex = fs.readFileSync(__dirname + "/__regex__/as_stop_response.regex").toString();
            const response = runCliScript(__dirname + "/__scripts__/address-space/as_fully_qualified.sh",
            TEST_ENVIONMENT_NO_PROF,
                [
                    DEFAULT_SYSTEM_PROPS.tso.account,
                    DEFAULT_SYSTEM_PROPS.zosmf.host,
                    DEFAULT_SYSTEM_PROPS.zosmf.port,
                    DEFAULT_SYSTEM_PROPS.zosmf.user,
                    DEFAULT_SYSTEM_PROPS.zosmf.pass
                ]);
            expect(response.stderr.toString()).toBe("");
            expect(response.status).toBe(0);
            expect(new RegExp(regex, "g").test(response.stdout.toString())).toBe(true);
        });
    });
});
