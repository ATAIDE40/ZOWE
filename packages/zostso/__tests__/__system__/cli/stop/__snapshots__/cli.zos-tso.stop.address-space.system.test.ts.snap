// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zos-tso start address-space should display the help 1`] = `
"
 COMMAND NAME
 ------------

   address-space | as

 DESCRIPTION
 -----------

   Start a TSO address space, from which you will receive a token (a.k.a
   'servelet-key') for further address space interaction (e.g. termination).

 USAGE
 -----

   bright zos-tso start address-space [options]

 PROFILE OPTIONS
 ---------------

   --zosmf-profile  | --zosmf-p (string) (optional)

      The name of a (zosmf) profile to load for this command execution.

   --tso-profile  | --tso-p (string) (optional)

      The name of a (tso) profile to load for this command execution.

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (flag) (optional)

      Produce the command response as a JSON document

   --help  | -h (flag) (optional)

      Display help text


 EXAMPLES
 --------

   Start TSO/E address space:

	 bright zos-tso start address-space 

   Start TSO/E address space, and receive response in JSON format:

	 bright zos-tso start address-space --rfj

{
  \\"success\\": true,
  \\"message\\": \\"The help was constructed for command: address-space.\\",
  \\"stdout\\": \\"\\\\n COMMAND NAME\\\\n ------------\\\\n\\\\n   address-space | as\\\\n\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Stop a TSO address space, from which you previously started and received a token\\\\n   (a.k.a 'servlet-key').\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   bright zos-tso stop address-space <servletkey> [options]\\\\n\\\\n POSITIONAL OPTIONS\\\\n ------------------\\\\n\\\\n   servletkey\\\\t\\\\t (string)\\\\n\\\\n      The servlet key from a previously started TSO address space.\\\\n\\\\n PROFILE OPTIONS\\\\n ---------------\\\\n\\\\n   --zosmf-profile  | --zosmf-p (string) (optional)\\\\n\\\\n      The name of a (zosmf) profile to load for this command execution.\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (flag) (optional)\\\\n\\\\n      Produce the command response as a JSON document\\\\n\\\\n   --help  | -h (flag) (optional)\\\\n\\\\n      Display help text\\\\n\\\\n\\\\n EXAMPLES\\\\n --------\\\\n\\\\n   Stop the TSO address space identified by IBMUSER-329-aafkaaoc:\\\\n\\\\n\\\\t bright zos-tso stop address-space IBMUSER-329-aafkaaoc\\\\n\\\\n\\",
  \\"stderr\\": \\"\\",
  \\"data\\": \\"\\\\n COMMAND NAME\\\\n ------------\\\\n\\\\n   address-space | as\\\\n\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Stop a TSO address space, from which you previously started and received a token\\\\n   (a.k.a 'servlet-key').\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   bright zos-tso stop address-space <servletkey> [options]\\\\n\\\\n POSITIONAL OPTIONS\\\\n ------------------\\\\n\\\\n   servletkey\\\\t\\\\t (string)\\\\n\\\\n      The servlet key from a previously started TSO address space.\\\\n\\\\n PROFILE OPTIONS\\\\n ---------------\\\\n\\\\n   --zosmf-profile  | --zosmf-p (string) (optional)\\\\n\\\\n      The name of a (zosmf) profile to load for this command execution.\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (flag) (optional)\\\\n\\\\n      Produce the command response as a JSON document\\\\n\\\\n   --help  | -h (flag) (optional)\\\\n\\\\n      Display help text\\\\n\\\\n\\\\n EXAMPLES\\\\n --------\\\\n\\\\n   Stop the TSO address space identified by IBMUSER-329-aafkaaoc:\\\\n\\\\n\\\\t bright zos-tso stop address-space IBMUSER-329-aafkaaoc\\\\n\\\\n\\"
}"
`;

exports[`zos-tso start address-space should throw an error if provided address space is inactive 1`] = `
"Command Error:
Expect Error: IZUG1126E: z/OSMF cannot correlate the request for key \\"ZOSMFAD-55-aaakaaac\\" with an active z/OS application session.
"
`;

exports[`zos-tso start address-space should throw an error if servlet key parameter is not provided 1`] = `
"Command Error:
Expect Error: IZUG1126E: z/OSMF cannot correlate the request for key \\"ZOSMFAD-55-aaakaaac\\" with an active z/OS application session.
"
`;
